### ISSUE TITLE
Implement "What's new" feature 

### ISSUE THREAD

**Issue by lesleyjanenorton** [mozilla-mobile/mozilla-vpn-client#1529]
<img width="140" alt="Screen Shot 2021-08-10 at 7 22 10 PM" src="https://user-images.githubusercontent.com/22355127/128951554-a47f5363-9c18-4283-bbcb-749882fb4858.png">

To support the "What's new" feature, we need to build a new UI tour component. 

**Figma**
https://www.figma.com/file/SjWR5U9aqEiUnmrkuS74O5/What's-new%3F-(Release-notes)?node-id=69%3A2931

**Previously implemented SwipeView example**
https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/ui/views/ViewOnboarding.qml

**Helpful Qt Docs**
https://doc.qt.io/qt-5/qml-qtquick-controls2-popup.html#popup-sizing
https://doc.qt.io/qt-5/qml-qtquick-controls2-swipeview.html

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-857)

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Verified this across platforms, while using the last VPN version (2.5.0 (2.202108310209)) and I have the following questions:

  * when is the component displayed automatically on top of the VPN activation screen and when is seen as a red bullet near settings gear and needs to be accessed manually?
  * the version of the VPN is not mentioned in the first page of the tour, as there is in above screenshot, intended?
  * cards are not vertically centered;
!image-20210831-091634.png|width=308,height=551!
  * “What’s new“ text highlights in light purple - expected?
  * After navigating the tour, the red bullet is not dissmissed from “What’s new” menu and user needs to navigate away from Settings page to dismiss it.

Thanks in advance!

**Comment by flozia** [comment]
Thank you @ValentinaPC — regarding you questions:

1. **When is the component displayed automatically**
    1.1 **Popup**
    The component should be displayed automatically to the user if there are new features. Once the user starts the tour or dismisses the popup it should not be displayed until there are new features introduced.
    
    1.2 **Unseen features indicator (red dot)**
The indicator should be shown as long there are new features that are not seen. As soon as a user starts taking the tour (presses “Take the tour”) from the popup or the “What’s new” view all currently new features are marked as “seen”.

2. **Version of the VPN is not mentioned**
Yes — that’s intended. The spec for the [initial page (Figma)](https://www.figma.com/file/SjWR5U9aqEiUnmrkuS74O5/What-s-new-Release-notes?node-id=134%3A4834) changed.

3. **Cards are not vertically centered**
Just fixed this in #1662

4. **“What’s new“ text highlights in light purple**
Yes — the purple highlight is expected if it looks like the following screenshot ([Figma](https://www.figma.com/file/SjWR5U9aqEiUnmrkuS74O5/What-s-new-Release-notes?node-id=296%3A7210)) this:
![tour-hover](https://user-images.githubusercontent.com/13835474/131493729-85fbc435-e9e5-4cfc-9461-59cb606eb9a1.png)

5. **Red bullet is not dismissed from “What’s new” menu**
That’s a bug and should be also fixed in #1662

**Comment by ValentinaPC** [comment]
@flozia Thanks for the detailed response.
Checking different languages we observed that in some of the langs where the string is translated the "?" was added at the end. I expect to see that in English also. What do you think?

**Comment by flozia** [comment]
> in some of the langs (German) where the string is translated the "?" was added at the end. I expect to see that in English also. What do you think?

@ValentinaPC I’m not sure about the first bullet point. What are your thoughts regarding the question mark @chrisrooneys and @peterbenvenuto?

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

This is how it looks like on iOS, last VPN 2.5.0 build. It is somehow expected, because the Contact Support is not yet finalized, and DNS is not yet implemented. But, in my opinion, we should show a general message in that screen, when no new features are available.

!20210907_141548.jpg|width=3024,height=4032!

**Comment by flozia** [comment]
@ValentinaPC I have a question to have more context regarding the following issue you reported in a previous comment:
> The string is not update after language switch only after navigating away from Settings screen

Is this just happening for the “What’s new” menu string or does this affect other strings as well? In the preferences I’m observing the same thing for the “Data collection and use” string.
<img width="472" alt="preferences-screenshot" src="https://user-images.githubusercontent.com/13835474/132411622-9803d6b8-dc26-4f48-a62a-e239c2a48d7b.png">

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Florian Zia Yes, I just saw this behavior for the “What’s new“ option.

For example, change the language from German to Spanish and navigate to Settings again → you will see the What’s new menu still in German. The Data collection is in Spanish, or the rest of the options. The What’s new, also changes eventually, if going out to main activation screen.

**Comment by flozia** [comment]
Thank you @ValentinaPC for checking this again. It seems like this may not be an issue with this feature itself — I opened this issue #1770 to investigate this further.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Verified this on the last VPN 2.5 2.202109091821 build on iOS 14.4.

If there are no new features, the “What’s new“ menu is not displayed.

For the language part, we will log a separate issue.

**Comment by data-sync-user** [comment]
➤ Luminita Buruian commented:

Closing this as it has already been verified.

Thank you.

Valentina Virlics
